Add v15 breaking changes section (#6530)
umbraco/UmbracoDocs: 15/umbraco-cms/fundamentals/setup/upgrading/version-specific/README.md

<details>

<summary>Umbraco 15</summary>

* **Snapshots are removed**

> Snapshots has been removed, this means that any code that used `IPublishedSnapshot`, and by extension `IPublishedSnapshotAccessor`, must be updated. Inject `IPublishedContentCache` or `IPublishedMediaCache` and use those directly instead.

Reviewer: ```suggestion
Snapshots have been removed, meaning any code using `IPublishedSnapshot`, and by extension `IPublishedSnapshotAccessor`, must be updated. Inject `IPublishedContentCache` or `IPublishedMediaCache` and use those directly instead.
```

* **Modelsbuilder models needs to be rebuilt**

> Models generated by ModelsBuilder used the `IPublishedSnapshot` interface, which has been removed. This means that the models needs to be rebuilt, the approach to this will differ on the models mode chosen:

Reviewer: ```suggestion
Models generated by ModelsBuilder used the `IPublishedSnapshot` interface, which has been removed. This means that the models need to be rebuilt. The approach to this will differ depending on the mode chosen:
```

> * InMemoryAuto - Remove the `umbraco\Data\TEMP\InMemoryAuto` folder, this will trigger a rebuild of the models.
> * SourceCodeAuto and SourceCodeManual - Remove the old models, by default these are located in the `\umbraco\models` folder, and then trigger a rebuild of the models from the dashboard in the backoffice.

Reviewer: ```suggestion
* InMemoryAuto - Remove the `umbraco\Data\TEMP\InMemoryAuto` folder to trigger a rebuild of the models.
* SourceCodeAuto and SourceCodeManual - Remove the old models located in the `\umbraco\models` folder by default. Trigger a rebuild of the models from the ModelsBuilder dashboard in the backoffice.
```
